swagger.yaml

```diff
@@ -0,0 +1,493 @@
+
+tags:
+  - name: upload
+    description: Endpoints for uploading collection instruments
+  - name: link-exercise
+    description: Endpoints for linking collection instruments to collection exercises
+  - name: unlink-exercise
+    description: Endpoints for unlinking collection instruments from collection exercises
+  - name: download-csv
+    description: Endpoints for downloading CSV files
+  - name: download
+    description: Endpoints for downloading collection instruments
+  - name: instrument-size
+    description: Endpoints for getting the size of a collection instruments
+  - name: collection-instrument
+    description: Endpoints for getting collection instruments
+  - name: info
+    description: Service information endpoints
+
+paths:
+  "/collection-instrument-api/1.0.2/upload/{exercise_id}/{RU_id}":
+    post:
+      summary: Encrypt and upload collection instrument associated to collection exercise and RU
+      tags:
+        - upload
+      parameters:
+        - in: path
+          name: exercise_id
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: 'fb2a9d3a-6e9c-46f6-af5e-5f67fec3c040'
+          description: The collection exercise ID.
+        - in: path
+          name: RU_id
+          required: true
+          schema:
+            type: string
+            format: uuid
+            example: 'test_ru_ref'
+          description: The RU reference
+        - in : query
+          name: file
+          required: true
+          schema:
+            type: string
+            example: 'file_name.xlsx'
+          description: Spreadsheet file
+      responses:
+        '200':
+          description: Successfully uploaded collection instrument
+          content: "The upload was successful"
+        '500':
+          description: Failed to publish upload message
+  "/collection-instrument-api/1.0.2/upload/{exercise_id}":
+    post:
+      summary: Encrypt and upload collection instrument associated to collection exercised without RU
+      tags:
+        - upload
+      parameters:
+        - in: path
+          name: exercise_id
+          required: true
+          schema:
+            type: string
+            format: uuid
+          example: 'fb2a9d3a-6e9c-46f6-af5e-5f67fec3c040'
+        - in: query
+          name: file
+          required: true
+          schema:
+            type: string
+            example: 'file_name.xlsx'
+          description: Spreadsheet file
+        - in: query
+          name: classifiers
+          required: true
+          schema:
+            type: string
+            format: dictionary
+            example: '{"FORM_TYPE":%20"001"}'
+          description: Collection instrument classifiers
+      responses:
+        '200':
+          description: Successfully uploaded collection instrument
+          content: "The upload was successful"
```

Comment:
This is not quite right. Whats the content type? `application/json`? `text/plain`? If its JSON you need to supply a schema. its not enought to just have `content: "some string"`
look at the API.yaml in the other repos for examples.

Reply:
I tried looking at the other API.yaml files, but I couldn't find any endpoints where only a message is returned. I have modified these `content:` tags to `text/plain` objects.